### PR TITLE
Set current user

### PR DESCRIPTION
**Pull request by elijahkim**
- [x] set current user
- [x] handle case of when a token is not provided
- [x] handle case of when the token is invalid
- [x] extract out json web token module 
- [x] clean up

### CODE REVIEW

File: test/plugs/current_user_test.exs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  test "plug assigns a guest user when a token is not provided" do
+    conn = conn(:get, "/locations")
+    opts = SpeakEasyApi.Plugs.CurrentUser.init(repo: Repo)
+    result = SpeakEasyApi.Plugs.CurrentUser.call(conn, opts)
+
+    assert result.assigns.current_user == %SpeakEasyApi.Guest{}
+  end
+
+  test "plug assigns nil to current user if an invalid token is provided" do
+    conn = conn(:get, "/locations")
```
Comment (cjbell) [inline test/plugs/current_user_test.exs]: I think a cleaner syntax here might be:

```elixir
conn = 
  conn(:get, "/locations")
  |> put_req_header("authorization", "bearer YXNkZmFzZGZhc2Rm")
```

Comment (elijahkim) [inline test/plugs/current_user_test.exs]: ok!

File: web/plugs/current_user.ex [hunk 2]
```diff
@@ -0,0 +1,39 @@
+defmodule SpeakEasyApi.Plugs.CurrentUser do
+  import Plug.Conn
+  alias SpeakEasyApi.User
```
Comment (cjbell) [inline web/plugs/current_user.ex]:
```elixir
alias SpeakEasyApi.Guest
alias SpeakEasyApi.JWTConverter
```

File: test/plugs/current_user_test.exs [hunk 3]
```diff
@@ -0,0 +1,37 @@
+defmodule SpeakEasyApi.CurrentUserTest do
+  use ExUnit.Case, async: true
+  use Plug.Test
+  alias SpeakEasyApi.User
+  alias SpeakEasyApi.Repo
```
Comment (cjbell) [inline test/plugs/current_user_test.exs]:
```elixir
alias SpeakEasyApi.Plugs.CurrentUser, as: Subject
```

**Pull request merged**